lib/network/modules/components/Edge.js

```diff
@@ -677,16 +686,73 @@ class Edge {
         // Ignore the orientations.
         this.labelModule.pointToSelf = true;
         var x, y;
-        var radius = this.options.selfReferenceSize;
-        if (node1.shape.width > node1.shape.height) {
+
+        var radius = this.options.selfReference.size;
+        var percentage = 0.125;
+        
+        // get circle coordinates
+        if (
+          typeof this.options.selfReference.angle !== "undefined" && this.options.selfReference.angle !== null
+        ) {
+          x = node1.x;
+          y = node1.y;
+          
+          if(ctx){
+            //calculating opposite and adjacent
+            //distaneToBorder becomes Hypotenuse. 
+            //Formulas sin(a) = Opposite / Hypotenuse and cos(a) = Adjacent / Hypotenuse
+            const toBorderDist = node1.distanceToBorder(ctx, this.options.selfReference.angle);
+            const sin = Math.sin(this.options.selfReference.angle)*toBorderDist;
+            const cos = Math.cos(this.options.selfReference.angle)*toBorderDist;
```

Comment:
Isn't `sin` the x distance from the center of the node? Wouldn't something like `xFromNodeCenter` be more descriptive?
dtto for `cos` on y axis.